simulators/ethereum/sync/sync.go

```diff
@@ -73,18 +88,68 @@ func runSourceTest(t *hivesim.T, c *hivesim.Client) {
 		Description: fmt.Sprintf("This test attempts to sync the chain from a %s node.", source.Type),
 		Parameters:  sinkParams,
 		Files:       sinkFiles,
-		Run:         runSyncTest,
+		Run:         runETHSyncTest,
 	})
 }
 
-func runSyncTest(t *hivesim.T, c *hivesim.Client) {
+func runETHSyncTest(t *hivesim.T, c *hivesim.Client) {
 	node := &node{c}
 	err := node.checkSync(t, testchainHeadNumber, testchainHeadHash)
 	if err != nil {
 		t.Fatal("sync failed:", err)
 	}
 }
 
+func runLESSourceTest(t *hivesim.T, c *hivesim.Client) {
+	// Short circuit if the source is not Geth client.
+	if !isGeth(c.Type) {
+		return
```

Comment:
I think this should be done a bit differently. If you use `ClientTestSpec`, it will launch an instance of the test for all client types. But this test will only work with geth. What you should to instead is:

```
sim := hivesim.New()
clientTypes, err := sim.ClientTypes()
if err != nil {
    panic(err)
}

suite := hivesim.Suite{ ... }
suite.AddTest(hivesim.ClientTestSpec{ ... eth test ... })

// Add the light client sync test. This only works with geth for now.
var gethClients []string
for _, name := range clientTypes {
    if strings.HasPrefix(name, "go-ethereum") {
        gethClients = append(gethClients, name)
    }
}
for _, name := range gethClients {
    name := name
    suite.AddTest(hivesim.TestSpec{
         Name: fmt.Sprintf("%s as LES server", c),
         Run: func (t *T) { runLESTest(t, name) },
    })
}
```

Reply:
Then, in `runLESTest`, you can use `t.RunClient` with all geth client types instead of using `t.RunAllClients`.

Reply:
Sorry, this is a bit complicated right now. We can make this simpler in a future version of package hivesim.